src/python/pants/binaries/binary_tool.py

```diff
@@ -115,6 +166,22 @@ def register_options(cls, register):
       version_registration_kwargs['fingerprint'] = True
     register('--version', **version_registration_kwargs)
 
+    register('--version-digest-mapping', type=dict,
+             default={
+               # "Serialize" the default value dict into "basic" types that can be easily specified
+               # in pants.ini.
+               platform_constraint.value: tool.into_tuple()
+               for platform_constraint, tool in cls.default_versions_and_digests.items()
+             },
+             fingerprint=True,
+             help='A dict mapping <platform constraint> -> (<version>, <fingerprint>, <size_bytes>).'
```

Comment:
Can you think of a usecase for having a different version per platform? Feels like that could safely stay in the `--version` field...

Reply:
If one of them needs to be patched for some platform-specific performance regression or security vulnerability!